pkg/traceql/ast.go

```diff
@@ -18,7 +18,8 @@ type metricsFirstStageElement interface {
 	Element
 	extractConditions(request *FetchSpansRequest)
 	init(req *tempopb.QueryRangeRequest, mode AggregateMode)
-	observe(Span)                        // TODO - batching?
+	observe(Span) // TODO - batching?
+	observeWithExemplar([]byte, Span)
```

Comment:
nit:  would flip the order of params to keep Span first.  Or since this causes a lot of overlapping functionality, go back to one method `observe(span, []byte)` and make the exemplar optional.